src/pages/utilities/animations.md

```diff
@@ -0,0 +1,372 @@
+---
+previousText: ''
+previousUrl: ''
+nextText: 'Config'
+nextUrl: '/docs/utilities/config'
+---
+
+# Animations
+
+## Overview
+
+Ionic Animations is a utility that allows developers to build complex animations in a platform agnostic manner. Developers do not need to be using a particular framework such as React or Angular, nor do they even need to be building an Ionic app. As long as developers have access to `@ionic/core`, they will have access to all of Ionic Animations.
+
+Building efficient animations can be tricky. Developers are often limited by the libraries available to them as well as the hardware that their apps run on. On top of that, many animation libraries use a JavaScript-driven approach to running animations where they handle the calculation of your animation's values at every step in a `requestAnimationFrame` loop. This reduces the scalability of your animations as the library is constantly computing values and using up CPU time.
+
+Ionic Animations uses the [Web Animations API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API) to build and run your animations. In doing this, we offload all work required to compute and run your animations to the browser. As a result, this allows the browser to make any optimizations it needs and ensures your animations run as smoothly as possible. While most browsers support a basic implementation of Web Animations, we fallback to [CSS Animations](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Animations/Using_CSS_animations) for browsers that do not support Web Animations. The performance difference in switching between these two should typically be negligible.
+
+## Basic Animations
+
+### Usage
+
+```javascript
+createAnimation()
+  .addElement(document.querySelector('.square'))
+  .duration(1500)
+  .iterations(Infinity)
+  .fromTo('transform', 'translateX(0px)', 'translateX(100px)')
+  .fromTo('opacity', 1, 0.2);
+```
+
+In the example above, an animation that changes the transform and opacity on the `.square` element has been created. This animation will run an infinite number of times, and each iteration of the animation will last 1500ms.
+
+<docs-codepen user="ionic" slug="bGbMojP"></docs-codepen>
+
+## Keyframe Animations
+
+Ionic Animations allows you to control the intermediate steps in an animation using keyframes. Any valid CSS property can be used here, and you can even use CSS Variables as values.
+
+### Usage
+
+```javascript
+createAnimation()
+  .addElement(document.querySelector('.square'))
+  .duration(3000)
+  .iterations(Infinity)
+  .keyframes([
+    { offset: 0, background: 'red' },
+    { offset: 0.72, background: 'var(--background)' },
+    { offset: 1, background: 'green' }
+  ]);
+```
+
+In the example above, the `.square` element will transition from a red background color, to a background color defined by the `--background` variable, and then transition on to a green background color.
```

Comment:
Am I understanding this one correctly? either 1) Each transition (offset "step"?) occurs once every 3000ms? 2) Or perhaps, the entire animation takes place in 3000ms? (looking at the codepen, it seems like #2 but it'd be good to clarify)

Reply:
The entire animation takes place in 3000ms. The offset is just the point during that 3000 that an animation change should occur. So in this case at offset 0.72 (3000 * 0.72 = 2160ms), the background should be whatever is defined in `var(--background)`.